### PR TITLE
[NO JIRA] Add `closeOnScrimClick` prop to modal and scrim components

### PR DESCRIPTION
**Pull request by raullm7**

### CODE REVIEW

File: packages/bpk-component-modal/src/BpkModalDialog.js [hunk 1]
```diff
@@ -66,17 +66,18 @@ const BpkModalDialog = props => {
...
-              {props.closeText}
-            </BpkButtonLink>
-          ) : (
-            <BpkCloseButton
-              className={getClassName('bpk-modal__close-button')}
-              label={props.closeLabel}
-              onClick={props.onClose}
-            />
-          )}
+          {props.closeOnScrimClick &&
```
Comment (matthewdavidson) [inline packages/bpk-component-modal/src/BpkModalDialog.js]: The behaviour of hiding the close button/link should be a separate prop from `closeOnScrimClick`.

File: packages/bpk-component-modal/src/BpkModal.js [hunk 2]
```diff
@@ -53,8 +54,9 @@ const BpkModal = props => {
       renderTarget={renderTarget}
     >
       <ScrimBpkModalDialog
-        onClose={onClose}
+        onClose={closeOnScrimClick ? onClose : () => {}}
```
Comment (matthewdavidson) [inline packages/bpk-component-modal/src/BpkModal.js]: The current implementation of triggering `onClose` when the scrim is clicked is in the `BpkScrim` component no? That means the `closeOnScrimClick` prop should be acted upon in that component as opposed to here.

File: packages/bpk-docs/src/pages/ModalsPage/ModalsPage.js [hunk 3]
```diff
@@ -131,6 +133,46 @@ const components = [
       </ModalContainer>,
     ],
   },
+  {
+    id: 'no-full-screen-on-mobile',
```
Comment (matthewdavidson) [inline packages/bpk-docs/src/pages/ModalsPage/ModalsPage.js]: We intentionally want to hide this example from the Modals docs page and only show it in the Dialog docs page (when it exists)

File: packages/bpk-component-modal/src/BpkModal.js (hunk 2 above)
Comment (raullm7) [reply packages/bpk-component-modal/src/BpkModal.js]: Done.

File: packages/bpk-component-modal/src/BpkModalDialog.js (hunk 1 above)
Comment (raullm7) [reply packages/bpk-component-modal/src/BpkModalDialog.js]: Done.

File: packages/bpk-docs/src/pages/ModalsPage/ModalsPage.js (hunk 3 above)
Comment (raullm7) [reply packages/bpk-docs/src/pages/ModalsPage/ModalsPage.js]: Done.

**Pull request merged**